### ISSUE TITLE
SC2183 only considers placeholders before the first real newline in format string

### ISSUE THREAD

**Issue by nwwdles** [koalaman/shellcheck#2007]
Hey! Thanks for this great piece of software, it really helps me a lot.

I had quite a long format string so I was using actual newlines instead of `\n`. I noticed that shellcheck issued a warning after I simply removed a newline from a format string.

#### For bugs
- Rule Id: `SC2183`
- My shellcheck version: `0.7.1` and online
- [x] The rule's wiki page does not already cover this (https://shellcheck.net/wiki/SC2183)
- [x] I tried on https://www.shellcheck.net/ and verified that this is still a problem on the latest commit
- [x] I searched through https://github.com/koalaman/shellcheck/issues and didn't find anything related

#### Here's a snippet or screenshot that shows the problem:

Case 1:

```sh
#!/bin/sh

printf '%s
%s' abc
```

Case 2:

```sh
#!/bin/sh

printf 'this way too
%s%s
%s' abc
```

#### Here's what shellcheck currently says:

Case 1: nothing
Case 2: `SC2183: This format string has 2 variables, but is passed 1 arguments.`

#### Here's what I wanted or expected to see:

Case 1: `SC2183: This format string has 2 variables, but is passed 1 arguments.`
Case 2 : `SC2183: This format string has 3 variables, but is passed 1 arguments.`

**Comment by koalaman** [comment]
Nice catch! This was due to ShellCheck's usage of regex to match formats like `%3.2s`, which included a `.*` at the end to get the rest of the string for further processing. However, the regex engine (Text.Regex.TDFA) stopped matching `.*` at line boundaries, effectively ignoring anything after the first literal linefeed. This has been fixed by explicitly matching `\n` along with `.`. Thanks!
